catalog/core/catalog-core-commons/src/main/java/ddf/catalog/filter/delegate/FilterToTextDelegate.java

```diff
@@ -20,6 +20,22 @@
 
 public class FilterToTextDelegate extends FilterDelegate<String> {
 
+    // Custom functions
+    @Override
+    public String propertyIsEqualTo(String functionName, List<Object> arguments, Object literal) {
+        switch (functionName) {
+        case "divisibleBy":
+            return propertyIsDivisibleBy((String) arguments.get(0), (Long) arguments.get(1)) + '='
+                    + literal;
+        case "proximity":
+            return propertyIsInProximityTo((String) arguments.get(0),
```

Comment:
Instead of doing the casting logic in every filter delegate I do it in the visit(Function) of the GeotoolsFilterAdapter. That being said right now if it can't coerce the value to the correct type it will just return it and would cause a casting error here( I can change it to throw an exception in the GeotoolsFilterAdapter).